src/main/java/org/team1540/elmo/utils/SignedSlewRateLimiter.java

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package org.team1540.elmo.utils;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.util.WPIUtilJNI;
+
+/**
+ * A class that limits the rate of change of an input value. Useful for implementing voltage,
+ * setpoint, and/or output ramps. A slew-rate limit is most appropriate when the quantity being
+ * controlled is a velocity or a voltage; when controlling a position, consider using a {@link
+ * edu.wpi.first.math.trajectory.TrapezoidProfile} instead.
+ */
+public class SignedSlewRateLimiter {
+  private final double m_rateLimitAcc;
+  private final double m_rateLimitDecc;
+  private double m_prevVal;
+  private double m_prevTime;
+
+  /**
+   * Creates a new SlewRateLimiter with the given rate limit and initial value.
+   *
+   * @param rateLimit The rate-of-change limit, in units per second.
+   * @param initialValue The initial value of the input.
+   */
+  public SignedSlewRateLimiter(double rateLimitAcc, double rateLimitDecc, double initialValue) {
+    m_rateLimitAcc = rateLimitAcc;
+    m_rateLimitDecc = rateLimitDecc;
+    m_prevVal = initialValue;
+    m_prevTime = WPIUtilJNI.now() * 1e-6;
```

Comment:
Uh, what's going on here?

Reply:
I just copied it from the WPILib SlewRateLimiter class. I just made it so that accelerating and decelerating has different limits, though i definitely could re-code the calculate() function more concisely. What are you worried about specifically with this bit?

Reply:
I was just confused about why you were calculating time like this. If that's how the real skew rate limiter works, I'm sure there's a reason for that

Reply:
dean kamen saw it in written in the magic dust
https://github.com/wpilibsuite/allwpilib/blob/b5a17f762c9877c2063e9ad73212ce59742983f9/wpimath/src/main/java/edu/wpi/first/math/filter/SlewRateLimiter.java#L36